### PR TITLE
Android: Add file parameter

### PR DESCRIPTION
**Pull request by m1ga**
Thanks to @appwerft for the ground work (https://github.com/hansemannn/titanium-firebase-core/issues/11)

**Test setup:**
* rename the google-services.json to gs.json
* place the file in `app/assets/android/gs.json`
* check the debug log for the `Firebase Options`
```
var fc = require('firebase.core');
fc.configure({
	file: "gs.json"
});
```

Debug message will also appear when you use the other parameter like this:
```
fc.configure({
	APIKey:"1",
	databaseURL:"2",
	projectID:"3",
	storageBucket: "4",
	applicationID: "5",
	GCMSenderID: "6"
});
```

### CODE REVIEW

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java [hunk 1]
```diff
@@ -36,25 +41,78 @@ public TitaniumFirebaseCoreModule()
...
+					}
+					if (projectInfo.has("project_number")) {
+						GCMSenderID = projectInfo.getString("project_number");
+					}
+					if (projectInfo.has("project_id")) {
+						projectID = projectInfo.getString("project_id");
+					}
+					if (json.has("client")) {
+						JSONArray clients = json.getJSONArray("client");
+						for (int i = 0,len = clients.length(); i < len; i++) {
```
Comment (hansemannn) [inline android/src/firebase/core/TitaniumFirebaseCoreModule.java]: missing space after the comma. Can you lint the source? We use clang-format for all native projects, e.g.
```bash
clang-format -style=file -i path/to/*.java
```

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java [hunk 2]
```diff
@@ -36,25 +41,78 @@ public TitaniumFirebaseCoreModule()
...
+				}
+				if (param.containsKey("applicationID")) {
+					applicationID = param.getString("applicationID");
+				}
+				if (param.containsKey("GCMSenderID")) {
+					GCMSenderID = param.getString("GCMSenderID");
+				}
 			}
+
+			Log.d(LCAT, "Firebase Options: " + apiKey+"|"+databaseURL+"|"+projectID+"|"+storageBucket+"|"+applicationID+"|"+GCMSenderID);
```
Comment (hansemannn) [inline android/src/firebase/core/TitaniumFirebaseCoreModule.java]: Do we still need this? I see a possible security issue with it.

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java (hunk 1 above)
Comment (m1ga) [reply android/src/firebase/core/TitaniumFirebaseCoreModule.java]: sure, made the changes, it changed quiet a lot that's why I didn't do it in the beginning

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java (hunk 2 above)
Comment (m1ga) [reply android/src/firebase/core/TitaniumFirebaseCoreModule.java]: no, was just for debug purpose. Removed it

**Comment by hansemannn** [comment]
Whoops, I forgot to say that we should use the `.clang-format` from `titanium_mobile/android/.clang-format` which is what's used in the SDK as well.

**Review by hansemannn — APPROVED** [review]
PR **approved**! The linting can be done later.

**Comment by hansemannn** [comment]
-> https://github.com/hansemannn/titanium-firebase-core/releases/tag/android-2.2.0

**Comment by m1ga** [comment]
Awesome, Thanks! I'll add that file to my future modules too! Would be nice to have that in the base-setup (if its not already added)

**Comment by hansemannn** [comment]
Yep, good idea! Can you do a PR for that? :D I may forget about it.

**Comment by m1ga** [comment]
sure https://github.com/appcelerator/titanium_mobile/pull/10015 ;)

**Pull request merged**